src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -728,6 +726,15 @@ void ClusterTaskManager::FillResourceUsage(rpc::ResourcesData &data) {
       by_shape_entry->set_backlog_size(backlog_it->second);
     }
   }
+
+  // Check whether resources have been changed.
+  auto last_heartbeat_resources = gcs_client_->NodeResources().GetLastResourceUsage();
```

Comment:
another note:
seems we already have last_heartbeat_resources in [NodeManager::FillResourceReport](https://github.com/ray-project/ray/blob/41cf2e3d50105a8bc42939e0bc32c6423bd95b55/src/ray/raylet/node_manager.cc#L557) before calling `cluster_task_manager.FillResourceUsage`. Does it make sense to reuse that and save a RPC call (and dependency here)?

Reply:
Actually `gcs_client_->NodeResources().GetLastResourceUsage()` doesn't send a RPC. It's just a getter method of a cache variable.

For saving the dependency of gcs_client, I think you are right. I'll modify it.